OSDOCS-3663: Various changes to ROSA prereqs (#46027)
openshift/openshift-docs: modules/rosa-sts-aws-requirements.adoc

[id="rosa-account_{context}"]
== Account

> * The customer ensures that the AWS limits are sufficient to support {product-title} provisioned within the customer's AWS account. Running `rosa verify quota` in the CLI validates that you have the required quota to run a cluster. See Additional resources for more information.

Reviewer: Can we also add
> Please note that quota verification takes into account your AWS quota, but does note compare your consumption versus your quota.
.. or similar

[id="rosa-account_{context}"]
== Account
* You must ensure that the AWS limits are sufficient to support {product-title} provisioned within the customer's AWS account. Running `rosa verify quota` in the CLI validates that you have the required quota to run a cluster.
+
[NOTE]
====

> Quota verification checks your AWS quota, but it does note compare your consumption versus your AWS quota. See the "Limits and scalability" link in the Additional resources for more information.

Reviewer 2: ```suggestion
Quota verification checks your AWS quota, but it does not compare your consumption to your AWS quota. See the "Limits and scalability" link in Additional resources for more information.
```

====
+
* If SCP policies are applied and enforced, these policies must not be more restrictive than the roles and policies required by the cluster.
* You may deploy native AWS services within the same AWS account.

> * Your account must have a service-linked role set up as it is required for elastic load balancers (ELBs) to be configured. See AWS Limits in the  Additional resources for information on creating a service-linked role for your ELB if you have not created a load balancer in your AWS account previously.

Reviewer 2: ```suggestion
* Your account must have a service-linked role set up as it is required for elastic load balancers (ELBs) to be configured. See AWS Limits in the  Additional resources for information about creating a service-linked role for your ELB if you have not created a load balancer in your AWS account previously.
```

[id="rosa-associating-account_{context}"]
== Associating your AWS account

> {product-title} (ROSA) cluster provisioning tasks require linked `ocm-role` and `user-role` {cluster-manager} IAM resources to your AWS account using your Amazon Resource Name (ARN).

Reviewer 3: s/cluster provisioning/cluster-provisioning ?? (I had to read it a few times to understand.)

s/linked/linking ?? require the `ocm-role` and `user-role` {cluster-manager} IAM resources be linked to   (Seems like there needs to be another verb somewhere.)

== Access requirements

* Red Hat must have AWS console access to the customer-provided AWS account. This access is protected and managed by Red Hat.

> * You must not use the AWS account to elevate their permissions within the {product-title} cluster.

Reviewer 2: ```suggestion
* You must not use the AWS account to elevate your permissions within the {product-title} cluster.
```

* Actions available in the `rosa` CLI utility or {cluster-manager-url} console must not be directly performed in the customer's AWS account.

> * You do not need to have a preconfigured domain to deploy ROSA clusters. If you wish to use a custom domain, see the Additional resources for information.

Reviewer 2: ```suggestion
* You do not need to have a preconfigured domain to deploy ROSA clusters. If you want to use a custom domain, see Additional resources for information.
```

== Security requirements
* Volume snapshots will remain within the customer's AWS account and customer-specified region.
* Red Hat must have ingress access to EC2 hosts and the API server from allow-listed IP addresses.

> * Red Hat must have egress allowed to the documented domains. See the "AWS firewall prerequisites" section for the designated domains.

Reviewer 4: If you split the module as suggested, you could even do `.Additional resources` links to sections farther down the page (I have had to do this for similar situations)
